core/src/main/scala/com/mesosphere/usi/core/SchedulerLogic.scala

```diff
@@ -0,0 +1,201 @@
+package com.mesosphere.usi.core
+
+import java.time.Instant
+
+import com.mesosphere.usi.core.models._
+import com.mesosphere.usi.models._
+
+/**
+  * Container class responsible for keeping track of the frame and cache.
+  *
+  * As a general rule, business logic does not manipulate the USI state directly, but rather does so by returning
+  * effects.
+  */
+private[core] class SchedulerLogic {
+  private var frame: Frame = Frame(Map.empty, Map.empty, Map.empty)
```

Comment:
Wdyt about moving this 3x `Map.empty, Map.empty, Map.empty` to the Frame's constructor as default values?

Reply:
Or `Frame.empty` similar to `FrameEffects.empty`?

Reply:
sure

Reply:
[ ] - DOING